include/deal.II/grid/reference_cell.h

```diff
@@ -783,11 +783,39 @@ namespace ReferenceCell
         {
           AssertIndexRange(face_no, n_faces());
 
-          if (face_no == 1)
+          if (face_no == 0)
             return ReferenceCell::Type::Quad;
           else
             return ReferenceCell::Type::Tri;
         }
+
+        unsigned int
+        face_to_cell_vertices(
+          const unsigned int  face,
+          const unsigned int  vertex,
+          const unsigned char face_orientation) const override
+        {
+#ifdef DEBUG
+          if (face == 0)
+            {
+              AssertIndexRange(vertex, 4);
+            }
+          else
+            {
+              AssertIndexRange(vertex, 3);
+            }
+#endif
```

Comment:
Why debug mode? `AssertIndexRange` is only evaluated in debug mode, isn't it?

Reply:
I can't remember if the compiler will complain about empty if-statements or not. If the CI passes then this is probably fine - I'll force push a version without the `#ifdef` check.